database/sqlite3.ksy

```diff
@@ -27,218 +27,476 @@ doc: |
+      - id: write_version
+        type: u1
+        enum: format_version
+        doc: File format write version
+      - id: read_version
+        type: u1
+        enum: format_version
+        doc: File format read version
+      - id: page_reserved_space_size
+        type: u1
+        doc: Bytes of unused "reserved" space at the end of each page. Usually 0.
+      - id: max_payload_fraction
+        type: u1
+        doc: Maximum embedded payload fraction. Must be 64.
+      - id: min_payload_fraction
+        type: u1
+        doc: Minimum embedded payload fraction. Must be 32.
+      - id: leaf_payload_fraction
+        type: u1
+        doc: Leaf payload fraction. Must be 32.
+      - id: file_change_counter
+        type: u4
+      - id: num_pages
+        type: u4
+        doc: Size of the database file in pages. The "in-header database size".
+      - id: first_freelist_trunk_page
+        type: freelist_trunk_page_pointer
+        doc: Page number of the first freelist trunk page.
+      - id: num_freelist_pages
+        type: u4
+        doc: Total number of freelist pages.
+      - id: schema_cookie
+        type: u4
+      - id: schema_format
+        type: u4
+        doc: The schema format number. Supported schema formats are 1, 2, 3, and 4.
+      - id: def_page_cache_size
+        type: u4
+        doc: Default page cache size.
+      - id: largest_root_page
+        type: u4
+        doc: The page number of the largest root b-tree page when in auto-vacuum or incremental-vacuum modes, or zero otherwise.
+      - id: text_encoding
+        type: u4
+        doc: The database text encoding. A value of 1 means UTF-8. A value of 2 means UTF-16le. A value of 3 means UTF-16be.
+      - id: user_version
+        type: u4
+        doc: The "user version" as read and set by the user_version pragma.
+      - id: is_incremental_vacuum
+        type: u4
+        doc: True (non-zero) for incremental-vacuum mode. False (zero) otherwise.
+      - id: application_id
+        type: u4
+        doc: The "Application ID" set by PRAGMA application_id.
+      - id: reserved_header_bytes
+        size: 20
+      - id: version_valid_for
+        type: u4
+      - id: sqlite_version_number
+        type: u4
+    instances:
+      page_size:
+        value: 'page_size_raw == 1 ? 0x10000 : page_size_raw'
+        doc: The database page size in bytes
+      usable_size:
+        value: 'page_size - page_reserved_space_size'
+        doc: The "usable size" of a database page
+      overflow_min_payload_size:
+        value: ((usable_size-12)*32/255)-23
+        doc: The minimum amount of inline b-tree cell payload
+      table_max_overflow_payload_size:
+        value: usable_size - 35
+        doc: The maximum amount of inline table b-tree cell payload
+      index_max_overflow_payload_size:
+        value: ((usable_size-12)*64/255)-23
+        doc: The maximum amount of inline index b-tree cell payload
```

Comment:
docstrings should be more precise regarding overflow

i would just copy the original descriptions

```diff
       overflow_min_payload_size:
         value: ((usable_size-12)*32/255)-23
-        doc: The minimum amount of inline b-tree cell payload
+        doc: The minimum amount of payload that must be stored on the btree page before spilling is allowed
       table_max_overflow_payload_size:
         value: usable_size - 35
-        doc: The maximum amount of inline table b-tree cell payload
+        doc: The maximum amount of payload that can be stored directly on the b-tree page without spilling onto an overflow page. Value for table page
       index_max_overflow_payload_size:
         value: ((usable_size-12)*64/255)-23
-        doc: The maximum amount of inline index b-tree cell payload
+        doc: The maximum amount of payload that can be stored directly on the b-tree page without spilling onto an overflow page. Value for index page
```

Reply:
I'm at the point where I've put way much more time in this than I was willing to. Feel free to take over from here and tinker on the definition.

Reply:
thanks for your work : )